src/bytes.ts

```diff
@@ -1,19 +1,12 @@
-import { bytesToHex } from 'ethereum-cryptography/utils';
+import type { BytesLike } from '@ethersproject/bytes';
+type HexString = string;
 
-export type Bytes = Uint8Array;
+import { arrayify as toBytes, hexlify as toHex, concat } from '@ethersproject/bytes';
 
-export function compareBytes(a: Bytes, b: Bytes): number {
-  const n = Math.min(a.length, b.length);
-
-  for (let i = 0; i < n; i++) {
-    if (a[i] !== b[i]) {
-      return a[i]! - b[i]!;
-    }
-  }
-
-  return a.length - b.length;
+function compare(a: BytesLike, b: BytesLike): number {
```

Comment:
Does the new function implement the same ordering as the previous one? At first sight I don't think it does. Wondering if this might cause a breaking change.

Reply:
I think the previous function was lexicographic ordering and the new one is numerical ordering.

However, it seems we always use this function on byte arrays of length 32, and in this domain the two functions would coincide.

Reply:
No, it doesn't, but it's related to this:

https://github.com/OpenZeppelin/merkle-tree/issues/30

We may mention it in the `CHANGELOG` but I don't think it's a significant change since people could only use the standardLeaf hash before

Reply:
> However, it seems we always use this function on byte arrays of length 32, and in this domain the two functions would coincide.

Yes this, "lexicographic"

![image](https://github.com/OpenZeppelin/merkle-tree/assets/33379285/48680019-5af2-4af0-a1b0-37c8faa4827a)

Reply:
I don't think it's necessary to mention it in the changelog. My concern was that it would result in tree leaves getting ordered differently. But I don't think this happens. Feel free to resolve if you agree.

Reply:
Its true that both function don't match in the general case, but they match in the case of 32 bytes buffers.

My concern with the previous one is that it doesn't clearly match the solidity version. If we ever get in a situation where both version order differently, my guess is that this new one will be the one that matches solidity's ordering.